runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -47,17 +52,32 @@ public Instance(
         this.importedFunctionsOffset = importedFunctionsOffset;
         this.importedTablesOffset = importedTablesOffset;
         this.memory = memory;
+        this.dataSegments = dataSegments;
         this.functions = functions.clone();
         this.types = types.clone();
         this.functionTypes = functionTypes.clone();
         this.imports = imports;
         this.machine = new Machine(this);
         this.tables = tables.clone();
         this.elements = elements.clone();
+
+        initialize();
+    }
+
+    private void initialize() {
```

Comment:
@danielperano this should enable you to implement the `GLOBAL_GET` constant expression needed by `data.wast`